src/libs/SidebarUtils.js

```diff
@@ -93,86 +91,44 @@ Onyx.connect({
 
-        // Skip this report if it has no participants and if it's not a type of report supported in the LHN
-        if (_.isEmpty(participants) && !isChatRoom && !isDefaultRoom && !isPolicyExpenseChat) {
-            return false;
-        }
-
-        const hasDraftComment = report.hasDraft || false;
-        const iouReport = report.iouReportID && iouReports && iouReports[`${ONYXKEYS.COLLECTION.REPORT_IOUS}${report.iouReportID}`];
-        const iouReportOwner = report.hasOutstandingIOU && iouReport
-            ? iouReport.ownerEmail
-            : '';
-
-        const reportContainsIOUDebt = iouReportOwner && iouReportOwner !== currentUserLogin;
-        const hasAddWorkspaceRoomError = report.errorFields && !_.isEmpty(report.errorFields.addWorkspaceRoom);
-        const shouldFilterReportIfEmpty = report.lastMessageTimestamp === 0
-
-            // We make exceptions for defaultRooms and policyExpenseChats so we can immediately
-            // highlight them in the LHN when they are created and have no messsages yet. We do
-            // not give archived rooms this exception since they do not need to be higlihted.
-            && !(!ReportUtils.isArchivedRoom(report) && (isDefaultRoom || isPolicyExpenseChat))
-
-            // Also make an exception for workspace rooms that failed to be added
-            && !hasAddWorkspaceRoomError;
-
-        const shouldFilterReportIfRead = hideReadReports && !ReportUtils.isUnread(report);
-        const shouldFilterReport = shouldFilterReportIfEmpty || shouldFilterReportIfRead;
-        if (report.reportID !== currentlyViewedReportID
-            && !report.isPinned
-            && !hasDraftComment
-            && shouldFilterReport
-            && !reportContainsIOUDebt) {
-            return false;
-        }
-
-        // We let Free Plan default rooms to be shown in the App, or rooms that also have a Guide in them.
-        // It's the two exceptions to the beta, otherwise do not show policy rooms in product
-        if (ReportUtils.isDefaultRoom(report)
-            && !Permissions.canUseDefaultRooms(betas)
-            && ReportUtils.getPolicyType(report, policies) !== CONST.POLICY.TYPE.FREE
-            && !ReportUtils.hasExpensifyGuidesEmails(lodashGet(report, ['participants'], []))) {
-            return false;
-        }
-
-        if (ReportUtils.isUserCreatedPolicyRoom(report) && !Permissions.canUsePolicyRooms(betas)) {
-            return false;
-        }
+    // Get all the display names for our reports in an easy to access property so we don't have to keep
+    // re-running the logic
+    const filteredReportsWithReportName = _.map(filteredReports, (report) => {
+        const personalDetailMap = OptionsListUtils.getPersonalDetailsForLogins(report.participants, personalDetails);
+        return {
+            ...report,
```

Comment:
NAB, but it might be good to avoid this spread here. If you have tons of reports I would imagine that cloning them all would be less efficient than something like storing the `reportDisplayNames` in a dictionary. Not sure how less efficient, but spread is one of those things you have to watch out for at scale.

Reply:
That's a good call out. Do you want to create a follow-up issue for that? That could probably help for squeaking out some more performance. With this PR, it was mostly just DRYing up the previous code which is why I didn't do too much modification.

Reply:
Oops, I just realized this was not moving existing code, it was new code that I wrote. I'll create the follow-up issue